packages/grid/x-grid/src/tests/events.XGrid.test.tsx

```diff
@@ -0,0 +1,267 @@
+    let ageColumnElement;
+    beforeEach(() => {
+      const handleClick = (params, event) => {
+        eventArgs = { params, event };
+      };
+      render(<TestEvents onColumnHeaderClick={handleClick} />);
+
+      ageColumnElement = getColumnHeaderCell(3);
+      fireEvent.click(ageColumnElement);
+    });
+
+    it('should contains the column header element', () => {
+      expect(eventArgs!.params.element).to.equal(ageColumnElement);
+    });
+    it('should contains the column def', () => {
+      expect(eventArgs!.params.colDef).to.equal(apiRef!.current.getColumnFromField('age'));
+    });
+    it('should contains the column index', () => {
+      expect(eventArgs!.params.colIndex).to.equal(2);
+    });
+    it('should contains the column field', () => {
+      expect(eventArgs!.params.field).to.equal('age');
+    });
+    it('should contains the api', () => {
+      expect(eventArgs!.params.api).to.equal(apiRef.current);
+    });
+  });
+
+  describe('RowsParams', () => {
+    let eventArgs: { params: GridRowParams; event: React.MouseEvent } | null = null;
+    let row2;
+    beforeEach(() => {
+      const handleClick = (params, event) => {
+        eventArgs = { params, event };
+      };
+      render(<TestEvents onRowClick={handleClick} />);
+
+      row2 = getRow(1);
+      fireEvent.click(row2);
+    });
+
+    it('should contains the row id', () => {
+      expect(eventArgs!.params.id).to.equal(2);
+    });
+    it('should contains the element', () => {
+      expect(eventArgs!.params.element).to.equal(row2);
+    });
+    it('should contains the row model', () => {
+      expect(eventArgs!.params.row).to.equal(baselineProps.rows[1]);
+    });
+    it('should contains the row index', () => {
+      expect(eventArgs!.params.rowIndex).to.equal(1);
+    });
+    it('should contains the columns', () => {
+      expect(eventArgs!.params.columns).to.equal(apiRef!.current.getAllColumns());
+    });
+    it('should contains the api', () => {
+      expect(eventArgs!.params.api).to.equal(apiRef.current);
+    });
+  });
+
+  describe('CellsParams', () => {
+    let eventArgs: { params: GridCellParams; event: React.MouseEvent } | null = null;
+    let cell22;
+    beforeEach(() => {
+      const handleClick = (params, event) => {
+        eventArgs = { params, event };
+      };
+      render(<TestEvents onCellClick={handleClick} />);
+
+      cell22 = getCell(1, 1);
+      fireEvent.click(cell22);
+    });
+
+    it('should contains the cell id', () => {
+      expect(eventArgs!.params.id).to.equal(2);
+    });
+    it('should contains the cell value', () => {
+      expect(eventArgs!.params.value).to.equal('Jack');
+    });
+
+    it('should consider value getter', () => {
+      const cellFirstAge = getCell(1, 3);
+      fireEvent.click(cellFirstAge);
+
+      expect(eventArgs!.params.value).to.equal('Jack_11');
+    });
+    it('should consider value formatter', () => {
```

Comment:
All these `it` adds to the test duration. It currently runs in 1.22s. Without them, it runs in 0.50s. Why not assert the whole structure in a deep equal?

Reply:
goal is to see exactly what fails in a test. So you know the issue directly

Reply:
Then a deep equal should do it. You should get a clear diff

Reply:
Oh no, I see, it's not equal we need here, hold on

Reply:
```jsx
    it('should contains the current values', () => {
      expect(eventArgs!.params).to.deep.include({
        id: 2,
        value: 'Jackk',
        formattedValue: 'Jack',
        isEditable: true,
        element: cell22,
        row: baselineProps.rows[1],
        rowIndex: 1,
        colDef: apiRef!.current.getColumnFromField('first'),
        colIndex: 1,
        api: apiRef.current,
      })
    });
```

<img width="1190" alt="Screenshot 2021-03-10 at 20 01 56" src="https://user-images.githubusercontent.com/3165635/110682709-87237180-81db-11eb-84af-5b32c1ca71d8.png">